### PR TITLE
ENT-6914 Upgrade to Gradle 7, Kotlin 1.8 and JDK 17 (OS) 

### PR DESCRIPTION
**Pull request by arshadm**
This PR upgrades this project for Corda 4.12.

Major changes
1. Gradle 7.6
2. Kotlin 1.8
3. JDK17

### CODE REVIEW

File: shell/src/integration-test/kotlin/net/corda/tools/shell/InteractiveShellIntegrationTest.kt [hunk 1]
```diff
@@ -598,4 +586,19 @@ class InteractiveShellIntegrationTest {
             waitForStateConsumption(stateRefs)
         }
     }
-}
\ No newline at end of file
+
+    @StartableByRPC
+    @InitiatingFlow
+    class FlowICanRun : FlowLogic<String>() {
```
Comment (adelel1) [inline shell/src/integration-test/kotlin/net/corda/tools/shell/InteractiveShellIntegrationTest.kt:592]: Why did you need to copy this class here from net.corda.tools.shell.SSHServerTest.FlowICanRun

Comment (arshadm) [reply shell/src/integration-test/kotlin/net/corda/tools/shell/InteractiveShellIntegrationTest.kt:592]: The test uses the enclosedCordapps functionality, so any cordapps used in the tests have to be in this class.

File: build.gradle [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import static org.gradle.api.JavaVersion.VERSION_1_8
+import static org.gradle.api.JavaVersion.VERSION_11
```
Comment (adelel1) [inline build.gradle]: Can you change this to VERSION_17, so now matches Corda

File: build.gradle [hunk 3]
```diff
@@ -146,53 +176,12 @@ allprojects {
     }
     tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).configureEach {
         kotlinOptions {
-            languageVersion = "1.2"
-            apiVersion = "1.2"
-            jvmTarget = VERSION_1_8
+            languageVersion = "1.8"
+            apiVersion = "1.8"
+            jvmTarget = VERSION_11
```
Comment (adelel1) [inline build.gradle]: Change to 17.

File: build.gradle (hunk 2 above)
Comment (arshadm) [reply build.gradle]: done

File: build.gradle (hunk 3 above)
Comment (arshadm) [reply build.gradle]: Done

File: shell/logs/node-icarus.log [hunk 4]
```diff
@@ -0,0 +1,35 @@
+[INFO ] 2023-12-05T15:48:15,422Z [Time-limited test] shell.HashLookupShellCommand. - Executing command "hashLookup". {}
```
Comment (adelel1) [inline shell/logs/node-icarus.log]: Here by mistake?

Comment (arshadm) [reply shell/logs/node-icarus.log]: yes, removed

**Review by adelel1 — APPROVED** [review]
LGTM
I am leaving to build team to review the Jenkins file.

**Review by wzur-r3 — APPROVED** [review]
LGTM for Jenkinsfile change
